src/cljs/rems/administration/create_workflow.cljs

```diff
@@ -47,28 +48,35 @@
  (fn [db [_ workflow]]
    (let [new-stuff {:title (:title workflow)
                     :organization (:organization workflow)
+                    :type (:type (:workflow workflow))
                     :handlers (mapv enrich-user (get-in workflow [:workflow :handlers]))}]
      (-> db
          (update ::form merge new-stuff)
          (dissoc ::loading-workflow?)))))
 
 ;;; form submit
 
+(def workflow-types #{:workflow/dynamic :workflow/bureaucratic :workflow/master})
+
+(defn needs-handlers? [type]
+  (contains? #{:workflow/dynamic :workflow/bureaucratic :workflow/master} type))
+
 (defn- valid-create-request? [request]
-  (and (case (:type request)
-         :auto-approve true
-         :dynamic (seq (:handlers request))
-         nil false)
-       (not (str/blank? (:organization request)))
-       (not (str/blank? (:title request)))))
+  (and
+   (contains? workflow-types (:type request))
+   (if (needs-handlers? (:type request))
+     (seq (:handlers request))
+     true)
+   (not (str/blank? (:organization request)))
+   (not (str/blank? (:title request)))))
 
 (defn build-create-request [form]
-  (let [request {:organization (trim-when-string (:organization form))
-                 :title (trim-when-string (:title form))
-                 :type (:type form)}
-        request (case (:type form)
-                  :auto-approve request
-                  :dynamic (assoc request :handlers (map :userid (:handlers form))))]
+  (let [request (merge
+                 {:organization (trim-when-string (:organization form))
+                  :title (trim-when-string (:title form))
+                  :type (:type form)}
+                 (when (needs-handlers? (:type form))
+                   {:handlers (map :userid (:handlers form))}))]
```

Comment:
This API needs updating at some point, perhaps make a task?

Reply:
In what way does it need to be updated? The structure of the :handlers list? The use of namespaced keywords?